### ISSUE TITLE
WP REST - PHP Notice: Trying to get property 'post_parent' of non-object error

### ISSUE THREAD

**Issue by lougreenwood** [10up/ElasticPress#1025]
**Have** you searched for similar issues before submitting this one?
Yes

**Is** this a bug, question or feature request?
Bug

**Describe the issue you encountered:**
When using this plugin alongside the `Members` plugin (https://wordpress.org/plugins/members/), any WP REST request to create a post returns an error:

```NOTICE: PHP message: PHP Notice: Trying to get property 'post_parent' of non-object in /var/www/html/htdocs/content/plugins/members/inc/template.php on line 95```

I initially reported the bug to that plugin author, but he suggested that this was a bug in another plugin. I then narrowed it down to ElasticPress after re-enabling plugins one-by-one.

Members bug report is here: https://wordpress.org/support/topic/trying-to-get-property-post_parent-of-non-object-in-template-php-on-line-95/#post-9982557

Current WordPress version:
4.9.4

Current ElasticPress version:
2.9.2

Current Elasticsearch version:
5.6.7

Where do you host your Elasticsearch server:
Local

Other plugins installed (WooCommerce, Simple Redirect Manager, etc..):
* Members
* Pods
* JSON Basic Auth

Steps to reproduce:

1. Using system which has the above plugins installed
2. Attempt to create a post to a CPT which is created by Pods using WP REST
3. See the error

For the diagnosis from the Member's plugin author, see this page: https://wordpress.org/support/topic/trying-to-get-property-post_parent-of-non-object-in-template-php-on-line-95/#post-9982557

**Comment by lougreenwood** [comment]
Also, forgot to mention that PHP version is 7.2.2

**Comment by nprasath002** [comment]
Hi @lougreenwood ,

We are working on a patch that may fix this issue.

**Comment by nprasath002** [comment]
Related https://github.com/10up/ElasticPress/issues/1023

**Comment by csloisel** [comment]
@lougreenwood This one is a bit of a grey area. The members plugin assumes that the `the_content` filter will only but applied inside of a post loop. It is true that usage of `the_content()` should only be used inside of a post loop, however we are applying the filter directly to the data for sync preparation and not using that template tag. Contextually, it doesn't seem to make a lot of sense to setup global post data or a post loop for this purpose.

I will start a discussion internally to decide if this is something we should address, and what the best way to go about that would be.

**Comment by ahmed-sigmalux** [comment]
What came of the internal discussion?

**Comment by allan23** [comment]
Hi @lougreenwood. I wanted to circle back to this issue and ask if you are still experiencing the error defined above.

I was unable to reproduce this issue locally using the latest versions of Pods and Members.
Steps I took:

1. Setup custom post type in Pods and enabled it in REST API.
2. Created a couple of posts in new post type.
2a. Ran a full re-index on section test.
3. Visited REST API endpoint.
